lib/host.rb

```diff
@@ -759,6 +778,42 @@ def copy_from(remote, local, **opts)
                    Host.localhost.absolutize(local, raw: opts[:raw])
     end
 
+    def apply_flags(other_hosts)
+      if self[:flags] == "/b/"
+        # setup bastion host
+        bastion = select_bastion(other_hosts)
+        if bastion.ssh_opts[:user]
+          bastion_connect_str = "#{bastion.ssh_opts[:user]}@#{bastion.hostname}"
+        else
+          bastion_connect_str = bastion.hostname
+        end
+
+        # setup proxy/jump/bastion host
+        bastion_key = bastion.ssh_opts[:private_key]
+        if bastion_key
+          key_str = "-i #{Host.localhost.shell_escape(expand_private_path(bastion_key))}"
+        end
+        self[:ssh_proxy] = "ssh -W %h:%p #{key_str} #{bastion_connect_str}"
+        # https://github.com/net-ssh/net-ssh/issues/653 jump lacks key setup
+        # self[:ssh_jump] = bastion_connect_str
+      elsif self[:flags] and !self[:flags].empty?
+        raise "flags '#{self[:flags]}' not supported by #{self}"
+      end
+    end
+
+    private def select_bastion(hosts)
+      hosts = hosts.select { |h| self.class === h }
+      b = hosts.find {|h| h.roles.include?(:bastion)}
+      unless b
+        # select first master as a bastion host
+        b = hosts.find { |h|
+          h.roles.include?(:master) && !h[:flags].include?("/b/")
```

Comment:
@akostadinov 
With :bastion set, `When I run commands on the host:` passed: https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/37541/console 
Without :bastion set, it failed: https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/37546/console :
```
      undefined method `include?' for nil:NilClass (NoMethodError)
      /home/jenkins/workspace/Runner-v3/lib/host.rb:810:in `block in select_bastion'
      /home/jenkins/workspace/Runner-v3/lib/host.rb:809:in `each'
      /home/jenkins/workspace/Runner-v3/lib/host.rb:809:in `find'
      /home/jenkins/workspace/Runner-v3/lib/host.rb:809:in `select_bastion'
      /home/jenkins/workspace/Runner-v3/lib/host.rb:784:in `apply_flags'
      /home/jenkins/workspace/Runner-v3/lib/environment.rb:433:in `host_add'
      /home/jenkins/workspace/Runner-v3/lib/openshift/node.rb:51:in `host'
      /home/jenkins/workspace/Runner-v3/features/step_definitions/node.rb:13:in `/^I select a random node's host$/'
      features/tierN/cli/service.feature:82:in `And I select a random node's host'
```

Reply:
fixed with #71